File: 2k-171/src/test/java/CalculatorInputParserTest.java
import org.junit.Assert;
import org.junit.Test;

import java.util.ArrayList;
import java.util.List;


public class CalculatorInputParserTest {

    private final AbstractParser parser = new CalculatorInputParser();

    @Test
    public void parse() {
        ArrayList<String> data = new ArrayList<>(List.of("2", "+", "2"));
        ArrayList<String> newData = parser.parse(data);
        Assert.assertArrayEquals(new ArrayList[]{data}, new ArrayList[]{newData});
    }

    @Test
    public void validateWithLessThanThreeArgs() {
        ArrayList<String> data = new ArrayList<>(List.of("2", "2"));
        Assert.assertThrows(IllegalArgumentException.class, () -> parser.validate(data));
    }

    @Test
    public void validateWithMoreThanThreeArgs() {
        ArrayList<String> data = new ArrayList<>(List.of("2", "+", "2", "+"));
        Assert.assertThrows(IllegalArgumentException.class, () -> parser.validate(data));
    }

    @Test
    public void validateWithoutFirstNumber() {
        ArrayList<String> data = new ArrayList<>(List.of("+", "2", "2"));
        Assert.assertThrows(IllegalArgumentException.class, () -> parser.validate(data));
    }

    @Test
    public void validateWithoutSecondNumber() {
        ArrayList<String> data = new ArrayList<>(List.of("2", "2", "+"));
        Assert.assertThrows(IllegalArgumentException.class, () -> parser.validate(data));
    }
}

File: 2k-227/src/test/kotlin/CalculatorInputParserTest.kt
import org.junit.Assert.*
import org.junit.Test
import kotlin.test.assertFailsWith

class CalculatorInputParserTest {

    private val parser: AbstractParser = CalculatorInputParser()

    @Test
    fun parse() {
        val data = ArrayList<String>(listOf("2", "+", "2"))
        val newData = parser.parse(data)
        assertArrayEquals(data.toArray(), newData.toArray())
    }

    @Test
    fun validateWithLessThanThreeArgs() {
        val data = ArrayList<String>(listOf("2", "2"))
        assertFailsWith<IllegalArgumentException> {
            parser.parse(data)
        }
    }

    @Test
    fun validateWithMoreThanThreeArgs() {
        val data = ArrayList<String>(listOf("2", "+", "2", "+"))
        assertFailsWith<IllegalArgumentException> {
            parser.parse(data)
        }
    }

    @Test
    fun validateWithoutFirstNumber() {
        val data = ArrayList<String>(listOf("+","2", "2"))
        assertFailsWith<IllegalArgumentException> {
            parser.parse(data)
        }
    }

    @Test
    fun validateWithoutSecondNumber() {
        val data = ArrayList<String>(listOf("2", "2", "+"))
        assertFailsWith<IllegalArgumentException> {
            parser.parse(data)
        }
    }
}